glnx-fdio.c

```diff
@@ -247,7 +245,7 @@ glnx_link_tmpfile_at (GLnxTmpfile *tmpf,
   const gboolean replace = (mode == GLNX_LINK_TMPFILE_REPLACE);
   const gboolean ignore_eexist = (mode == GLNX_LINK_TMPFILE_NOREPLACE_IGNORE_EXIST);
 
-  g_return_val_if_fail (tmpf->src_dfd >= 0, FALSE);
+  g_return_val_if_fail (tmpf->fd >= 0, FALSE);
```

Comment:
Another wrinkle here is that in various places we also take `-1` to mean `AT_FDCWD`; see `glnx_dirfd_canonicalize()`.  So...let me propose a variant of this patch which just uses an `initialized` member which should simplify things a lot.